[frio] Flip replyto link

xundeenergie opened the pull request:
Resolves #13524 and #13520

MrPetovan committed:
Normalize template variables in Widget\VCard

- Display new links in vier
- Address https://github.com/friendica/friendica/issues/13157#issuecomment-1751750581

MrPetovan committed:
Add callstack to log messages to better troubleshoot warning messages

MrPetovan committed:
Add a couple of missing fields in Content\Item->localize

- Address "Undefined array key "author-alias"" in Content\Item->localize:323

Commit:
make reply-to link depending on screen-width

* on small/smartphone-screens show "reply to" link above the posting and
  an eventual "shared by" link in a second line.
* on middle/tablet-screens show "reply-to" and "shared-by" link in one
  line above the posting, divided by a dot.
* on large screens show reply-to-link next to the time/date-link when
  posting was posted.

the first to positions are better for users with thick fingers on
smartphones. (like me)

Change "in reply to" to "is reply to". Sounds better.

Commit:
make reply-to link depending on screen-width

* on small/smartphone-screens show "reply to" link above the posting and
  an eventual "shared by" link in a second line.
* on middle/tablet-screens show "reply-to" and "shared-by" link in one
  line above the posting, divided by a dot.
* on large screens show reply-to-link next to the time/date-link when
  posting was posted.

the first to positions are better for users with thick fingers on
smartphones. (like me)

Change "in reply to" to "is reply to". Sounds better.

translations done

Commit:
set font-size of reply-to and shared-by to class time

Raroun commented:
Just a question:
Is it possible that "reply to" shows up on transifex to be translated? :)

xundeenergie commented:
> Just a question: Is it possible that "reply to" shows up on transifex to be translated? :)

It's made translateable, as all other text too, for tranisflex.

Commit:
show "parent unknown" when parent not federated

in some cases (visibility is private of parent comment), there is no
parent federated to my instance, and then the comment "hangs around"
with no information, why there is no parent.

For normal users this can be confusing, why all exept "this" comment
have parents.

So i added a simple text instead of link to parent with a "title" field
with a small explanation, why parent is not viewable.

translations done

xundeenergie commented:
The new text for comments without parents:

on small screen:
![Bildschirmfoto vom 2023-10-09 20-05-49](https://github.com/friendica/friendica/assets/5894829/8f14f56b-c296-4efb-bfad-afe0620c4eb1)

and on wide screen
![Bildschirmfoto vom 2023-10-09 20-06-33](https://github.com/friendica/friendica/assets/5894829/3ac41e71-db92-4fff-95df-bfb49d61ccde)

Commit:
Merge branch '2023.09-rc' into flip-replyto-link

Commit:
translations

xundeenergie commented:
Btw... this is such a comment
https://soc.schuerz.at/display/67d0f1d0-abb2c986d4e1ba42-5ddcaaba

And i don't know, why messages.po_check failed...

i merged 2023.09-rc into this branch, run bin/run_xgettext.sh and committed all.
Then i pushed to github.

What's wrong?

Commit:
Merge branch '2023.09-rc' into flip-replyto-link

Commit:
translations

MrPetovan commented:
Maybe another pull request was merged in the meantime, which triggers the merge conflict? In doubt, do the merge + script execution again.

Commit:
translation

Commit:
Merge branch '2023.09-rc' into flip-replyto-link

xundeenergie commented:
I've done the merge and commit again.

MrPetovan commented on src/Object/Post.php:
```diff
@@ -499,10 +499,13 @@ public function getTemplateData(array $conv_responses, string $formSecurityToken
 		$tmp_item = [
 			'parentguid'      => $parent_guid,
-			'isreplyto'       => DI::l10n()->t('in reply to %s', $parent_username),
+			'isreplyto'       => DI::l10n()->t('is reply to %s', $parent_username),
```
Please keep the `in reply to` translation. Additionally, please rename all `isreplyto` variables and array keys to `inreplyto` to match the translation string change.

xundeenergie replied:
Ok.
But why? 
It sounds strange to my ears...

xundeenergie replied:
i changed it and all occurances

MrPetovan replied:
Thanks, the (American-)English expression is "in reply to" as evidenced in the Mail header `In-Reply-To`. You could also say "This is a reply to" but it's longer and requires a subject that "in reply to" naturally implies.

MrPetovan commented on src/Object/Post.php:
```diff
@@ -499,10 +499,13 @@ public function getTemplateData(array $conv_responses, string $formSecurityToken
-			'isreplyto'       => DI::l10n()->t('in reply to %s', $parent_username),
+			'isreplyto'       => DI::l10n()->t('is reply to %s', $parent_username),
+			'isunknown'       => $parent_unknown,
+			'isunknown_label' => DI::l10n()->t('parent is probably private or not federated'),
```
```suggestion
			'isunknown_label' => DI::l10n()->t('Parent is probably private or not federated.'),
```

xundeenergie committed:
Update src/Object/Post.php

Sugfestion accepted

Co-authored-by: Hypolite Petovan <hypolite@mrpetovan.com>

Commit:
change isreplyto to inreplyto

Commit:
translations

Commit:
change last occurances isreplyto to inreplyto

MrPetovan commented on src/Object/Post.php:
```diff
@@ -499,10 +499,13 @@ public function getTemplateData(array $conv_responses, string $formSecurityToken
 
 		$parent_guid     = $thread_parent[$item['thr-parent-id']]['guid'] ?? '';
 		$parent_username = $thread_parent[$item['thr-parent-id']]['name'] ?? '';
+		$parent_unknown  = $parent_username ? '' : DI::l10n()->t('parent unknown');
```
```suggestion
		$parent_unknown  = $parent_username ? '' : DI::l10n()->t('Unknown parent');
```

xundeenergie committed:
Update src/Object/Post.php

Rename from 'parent unknown' to 'Unkknown parent"

Co-authored-by: Hypolite Petovan <hypolite@mrpetovan.com>

MrPetovan commented:
Thank you for the change, you will have to regenerate the messages.po file.

Commit:
run gettext
